gix/tests/gix/status.rs

```diff
@@ -310,6 +310,27 @@ mod index_worktree {
             );
         }
 
+        #[test]
+        fn submodule_in_symlinked_dir() -> crate::Result {
```

Comment:
This test should be removed in favor of specific tests which make clear what's under test.
This test 'only' tests if the status succeeds, otherwise it fails with something like.

```
Error: IndexAsWorktreeWithRenames(TrackedFileModifications(SubmoduleStatus { rela_path: "m1", source: SubmoduleStatus(NextStatusItem(IndexWorktree(IndexAsWorktreeWithRenames(DirWalk(WorktreeRootIsFile { root: "tests/fixtures/generated-do-not-edit/make_submodule_with_symlinked_git_dir/3844911589-unix/m1" }))))) }))
```